ironfish/src/rpc/routes/fees/getFees.ts

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import * as yup from 'yup'
+import { Assert } from '../../../assert'
+import { ApiNamespace, router } from '../router'
+
+export type GetFeesRequest = { numOfBlocks: number }
+export type GetFeesResponse = {
+  startBlock: number
+  endBlock: number
+  p25: number
+  p50: number
+  p75: number
+}
+
+export const GetFeesRequestSchema: yup.ObjectSchema<GetFeesRequest> = yup
+  .object({
+    numOfBlocks: yup.number().defined(),
+  })
+  .defined()
+
+export const GetFeesResponseSchema: yup.ObjectSchema<GetFeesResponse> = yup
+  .object({
+    startBlock: yup.number().defined(),
+    endBlock: yup.number().defined(),
+    p25: yup.number().defined(),
+    p50: yup.number().defined(),
+    p75: yup.number().defined(),
+  })
+  .defined()
+
+const percentile = (fees: number[], percentile: number): number => {
+  const pos = (fees.length - 1) * (percentile / 100)
+  const base = Math.floor(pos)
+
+  if (fees[base + 1]) {
+    const remainder = pos - base
+    return fees[base] + remainder * (fees[base + 1] - fees[base])
+  } else {
+    return fees[base]
+  }
+}
+
+router.register<typeof GetFeesRequestSchema, GetFeesResponse>(
+  `${ApiNamespace.fees}/getFees`,
+  GetFeesRequestSchema,
+  async (request, node): Promise<void> => {
+    const numOfBlocks = request.data.numOfBlocks - 1
```

Comment:
nit - not a huge fan of this, as it creates a misleadingly named var.

You use this var for `startBlock` below, and have a `<=` in the loop. 

Instead, you can do a `-1` for startBlock and a `<` for the loop.

Reply:
makes sense, Assert needs to be `-1` too